### PR TITLE
SLI-88 Error updating when a project no longer exists in SonarQube

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/core/ServerUpdateTask.java [hunk 1]
```diff
@@ -101,8 +103,18 @@ public void run(@NotNull ProgressIndicator indicator) {
   }
 
   private void updateModule(ConnectedSonarLintEngine engine, ServerConfiguration serverConfiguration, String key) {
-    engine.updateModule(serverConfiguration, key);
-    log.log("Module '" + key + "' in server binding '" + server.getName() + "' updated", LogOutput.Level.INFO);
+    try {
+      engine.updateModule(serverConfiguration, key);
+      log.log("Module '" + key + "' in server binding '" + server.getName() + "' updated", LogOutput.Level.INFO);
+    } catch (Exception e) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/core/ServerUpdateTask.java]: MAJOR Either log or rethrow this exception. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: src/main/java/org/sonarlint/intellij/core/SonarLintServerManager.java [hunk 2]
```diff
@@ -140,6 +142,9 @@ public void run() {
   }
 
   private SonarLintFacade createConnectedFacade(Project project, String serverId, String projectKey) {
```
Comment (SonarTech) [inline src/main/java/org/sonarlint/intellij/core/SonarLintServerManager.java]: MAJOR The Cyclomatic Complexity of this method "createConnectedFacade" is 12 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [SonarQubeServerMgmtPanel.java#L333](https://github.com/SonarSource/sonarlint-intellij/blob/40d8cde8b2a1fc759265f39515e376ff09f97bce/src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java#L333): The Cyclomatic Complexity of this method "actionUpdateServerTask" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MINOR [SonarLintFacade.java#L25](https://github.com/SonarSource/sonarlint-intellij/blob/40d8cde8b2a1fc759265f39515e376ff09f97bce/src/main/java/org/sonarlint/intellij/core/SonarLintFacade.java#L25): Remove this unused import 'org.jetbrains.annotations.Nullable'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [SonarLintProjectNotifications.java#L57](https://github.com/SonarSource/sonarlint-intellij/blob/40d8cde8b2a1fc759265f39515e376ff09f97bce/src/main/java/org/sonarlint/intellij/core/SonarLintProjectNotifications.java#L57): Define a constant instead of duplicating this literal "Please check the <a href='#'>SonarLint project configuration</a>" 3 times. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1192)

**Comment by henryju** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [SonarQubeServerMgmtPanel.java#L333](https://github.com/SonarSource/sonarlint-intellij/blob/8aae6debbcde2847374fc308fded5f59c4aa1070/src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java#L333): The Cyclomatic Complexity of this method "actionUpdateServerTask" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MINOR [SonarLintFacade.java#L25](https://github.com/SonarSource/sonarlint-intellij/blob/8aae6debbcde2847374fc308fded5f59c4aa1070/src/main/java/org/sonarlint/intellij/core/SonarLintFacade.java#L25): Remove this unused import 'org.jetbrains.annotations.Nullable'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [SonarLintProjectNotifications.java#L57](https://github.com/SonarSource/sonarlint-intellij/blob/8aae6debbcde2847374fc308fded5f59c4aa1070/src/main/java/org/sonarlint/intellij/core/SonarLintProjectNotifications.java#L57): Define a constant instead of duplicating this literal "Please check the <a href='#'>SonarLint project configuration</a>" 3 times. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1192)

**Pull request merged**